src/include/txn_inline.h

```diff
@@ -1227,6 +1227,8 @@ __wt_txn_begin(WT_SESSION_IMPL *session, const char *cfg[])
     if (F_ISSET(S2C(session), WT_CONN_READONLY))
         F_SET(txn, WT_TXN_READONLY);
 
+    WT_ASSERT(session, txn->mod_count == 0);
```

Comment:
Is it worth changing this WT_ASSERT to WT_ASSERT_ALWAYS

Reply:
Good idea - thank you.

Reply:
Done!